experiment/popularity_community/crawl_torrents.py

```diff
@@ -0,0 +1,214 @@
+        request = CrawlerSelectRequest(self.request_cache, hexlify(peer.mid), kwargs)
+        self.request_cache.add(request)
+        self.ez_send(peer, RemoteSelectPayload(request.number, json.dumps(kwargs).encode('utf8')))
+
+    def save(self):
+        self._logger.info(f"Save results to : {self._output_file_path}")
+
+        with open(self._output_file_path, 'w') as json_file:
+            json.dump(self.torrent_list, json_file)
+
+    def introduction_response_callback(self, peer, dist, payload):
+        """ This method adds logging behaviour to the base implementation
+        """
+        super().introduction_response_callback(peer, dist, payload)
+        self._logger.debug(f"Peer {peer} wants to introduce {payload.wan_introduction_address}")
+
+
+class Service(TinyTriblerService):
+    def __init__(self, output_file_path, timeout_in_sec, working_dir, config_path):
+        super().__init__(Service.create_config(working_dir, config_path),
+                         timeout_in_sec, working_dir, config_path)
+        self._output_file_path = output_file_path
+
+    @staticmethod
+    def create_config(working_dir, config_path):
+        config = TinyTriblerService.create_default_config(working_dir, config_path)
+
+        config.set_ipv8_enabled(True)
+        config.set_ipv8_walk_interval = IPV8_WALK_INTERVAL
+
+        config.set_chant_enabled(True)
+        return config
+
+    async def on_tribler_started(self):
+        await super().on_tribler_started()
+        session = self.session
+        peer = Peer(session.trustchain_keypair)
+
+        session.remote_query_community_community = Crawler(peer, session.ipv8.endpoint,
+                                                           session.ipv8.network,
+                                                           session.mds, self._output_file_path)
+
+        session.ipv8.overlays.append(session.remote_query_community_community)
+        session.ipv8.strategies.append((RandomWalk(session.remote_query_community_community,
+                                                   timeout=RANDOM_WALK_TIMEOUT_IN_SEC,
+                                                   window_size=RANDOM_WALK_WINDOW_SIZE,
+                                                   reset_chance=RANDOM_WALK_RESET_CHANCE),
+                                        UNLIMITED))
+
+
+def _parse_argv():
+    parser = argparse.ArgumentParser(description='Crawl first 100 torrents from random nodes in the network')
+    parser.add_argument('-t', '--timeout', type=int, help='a time in sec that the experiment will last', default=0)
+    parser.add_argument('-f', '--file', type=str, help='result file path (json)', default='result.json')
+    parser.add_argument('-v', '--verbosity', help='increase output verbosity', action='store_true')
+
+    return parser.parse_args()
+
+
+def _run_tribler(arguments):
+    service = Service(arguments.file, arguments.timeout,
+                      working_dir=os.path.join(
```

Comment:
Please use `pathlib` instead of `os` module. `pathlib` is much safer, modern and convenient.

Reply:
Ok, but give me proof first, please.

Reply:
https://treyhunner.com/2018/12/why-you-should-be-using-pathlib/

Reply:
@ichorid cool. Will fix it.

Reply:
fixed